Backend/mentii/user_ctrl.py

```diff
@@ -252,3 +251,51 @@ def getRole(userEmail, dynamoDBInstance):
       userRole = res['Item']['userRole']
 
   return userRole
+
+def joinClass(jsonData, dynamoDBInstance, email=None):
```

Comment:
The complexity of this function is pretty high. Do you think you could break out some of these if/else into other functions that you can just call?